### PR TITLE
Improve support for Gxxxx cost codes

### PR DESCRIPTION
**Pull request by NikolausC**

### CODE REVIEW

**Comment by NikolausC** [comment]
retest this please

File: app/models/node.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ class Node < ApplicationRecord
 
   scope :active, -> { where(deactivated_by: nil) }
 
-  scope :with_cost_code, -> { where(Node.arel_table[:cost_code].matches('S%')) }
+  scope :with_cost_code, -> { where(Node.arel_table[:cost_code].matches('S%').or(Node.arel_table[:cost_code].matches('G%'))) }
```
Comment (jbeast) [inline app/models/node.rb]: I think `where.not(cost_code: nil)` would be more maintainable 🚚

File: app/models/node.rb [hunk 2]
```diff
@@ -37,7 +35,7 @@ class Node < ApplicationRecord
 
   scope :active, -> { where(deactivated_by: nil) }
 
-  scope :with_cost_code, -> { where(Node.arel_table[:cost_code].matches('S%')) }
+  scope :with_cost_code, -> { where.not(cost_code: nil) }
```
Comment (pjvv) [inline app/models/node.rb]: Doesn't this allow any cost code now?

Comment (NikolausC) [reply app/models/node.rb]: No, this is for retrieving Nodes with cost codes from the database. Cost code format and general validation is handled when that data is added. Previously, this was essentially validating data on its way out of the database as well [@jbeast 2018]

**Pull request merged**